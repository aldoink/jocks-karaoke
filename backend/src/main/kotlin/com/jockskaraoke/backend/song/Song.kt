package com.jockskaraoke.backend.song

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "SONGS")
data class Song(
    @Id
    @GeneratedValue
    val id: Int,
    val location: String,
    val artist: String,
    val title: String
)