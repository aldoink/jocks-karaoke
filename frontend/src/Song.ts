export class Song {
    constructor(
        public location: string,
        public title: string,
        public artist: string) {}
}