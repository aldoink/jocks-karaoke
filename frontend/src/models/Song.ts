export class Song {
    constructor(
        public id: number,
        public location: string,
        public title: string,
        public artist: string
    ) {}
}