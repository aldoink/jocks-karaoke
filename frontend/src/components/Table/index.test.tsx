import React from 'react';
import renderer from 'react-test-renderer'
import {Song} from "../../Song";
import {Table} from "./index";

const entries = [
    new Song("BP1", "A Title", "A Test Artist"),
    new Song("BP2", "B Title", "B Test Artist"),
    new Song("BP3", "C Title", "C Test Artist"),
]

it('renders table entries', () => {
    const tree = renderer.create(<Table songList={entries}/>).toJSON();
    expect(tree).toMatchSnapshot();
});