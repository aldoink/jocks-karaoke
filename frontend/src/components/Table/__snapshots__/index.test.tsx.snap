// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table entries 1`] = `
<div
  className="table-container"
>
  <div
    className="table-cell bold"
  >
    Song Title
  </div>
  <div
    className="table-cell bold"
  >
    Artist
  </div>
  <div
    className="table-cell bold"
  >
    Location
  </div>
  <div
    className="table-cell"
  >
    A Title
  </div>
  <div
    className="table-cell"
  >
    A Test Artist
  </div>
  <div
    className="table-cell"
  >
    BP1
  </div>
  <div
    className="table-cell"
  >
    B Title
  </div>
  <div
    className="table-cell"
  >
    B Test Artist
  </div>
  <div
    className="table-cell"
  >
    BP2
  </div>
  <div
    className="table-cell"
  >
    C Title
  </div>
  <div
    className="table-cell"
  >
    C Test Artist
  </div>
  <div
    className="table-cell"
  >
    BP3
  </div>
</div>
`;
